pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -437,9 +436,10 @@ func constructServiceEntries(
 	res := make([]*workloadapi.Service, 0, len(svc.Spec.Hosts))
 	for _, h := range svc.Spec.Hosts {
 		// if we have no user-provided hostsAddresses and h is not wildcarded and we have hostsAddresses supported resolution
-		// we can try to use autoassigned hostsAddresses
+		// we can try to use autoassigned hostsAddresse
 		hostsAddresses := addresses
-		if len(hostsAddresses) == 0 && !host.Name(h).IsWildCarded() && svc.Spec.Resolution != v1alpha3.ServiceEntry_NONE {
+		// TODO(jaellio): conditionally block usage of wildcard usage based on other spec settings
```

Comment:
This is a blocker IMO

Reply:
Agree. This will somewhat be handled through the use of a new dns resolution type